File: 3959-check-good-integer/3959-check-good-integer.ex
defmodule Solution do
  @spec check_good_integer(n :: integer) :: boolean
  def check_good_integer(n) do
    solve(n, 0)
  end

  defp solve(0, sum), do: sum >= 50

  defp solve(n, sum) do
    digit = rem(n, 10)
    solve(div(n, 10), sum + digit * (digit - 1))
  end
end

File: 3959-check-good-integer/3959-check-good-integer.erl
-spec check_good_integer(N :: integer()) -> boolean().
check_good_integer(N) ->
    solve(N, 0).

solve(0, Sum) ->
    Sum >= 50;
solve(N, Sum) ->
    Digit = N rem 10,
    solve(N div 10, Sum + Digit * (Digit - 1)).